test/test_formatters.rb

```diff
@@ -47,26 +47,89 @@ def setup
+
+      '<p>para</p>'               => '<p>para</p>',
+      '<p dir="ltr">para</p>'     => '<p dir="ltr">para</p>',
+
+      # FIXME: Tests for escapes
+
       '<b>%@</b>'                 => '&lt;b>%s&lt;/b>',
+      '<em>%@</em>'               => '&lt;em>%s&lt;/em>',
+
       '<i>%@</i>'                 => '&lt;i>%s&lt;/i>',
+      '<cite>%@</cite>'           => '&lt;cite>%s&lt;/cite>',
+      '<dfn>%@</dfn>'             => '&lt;dfn>%s&lt;/dfn>',
+
+      '<big>%@</big>'             => '&lt;big>%s&lt;/big>',
+      '<small>%@</small>'         => '&lt;small>%s&lt;/small>',
+
+      '<font color="#45C1D0>%@</font>' => '&lt;font color="#45C1D0>%s&lt;/font>',
+
+      '<tt>%@</tt>'               => '&lt;tt>%s&lt;/tt>',
+
+      '<s>%@</s>'                 => '&lt;s>%s&lt;/s>',
+      '<strike>%@</strike>'       => '&lt;strike>%s&lt;/strike>',
+      '<del>%@</del>'             => '&lt;del>%s&lt;/del>',
+
       '<u>%@</u>'                 => '&lt;u>%s&lt;/u>',
 
-      '<span>inline</span>'       => '&lt;span>inline&lt;/span>',
-      '<p>paragraph</p>'          => '&lt;p>paragraph&lt;/p>',
+      '<super>%@</super>'         => '&lt;super>%s&lt;/super>',
+
+      '<sub>%@</sub>'             => '&lt;sub>%s&lt;/sub>',
+
+      '<ul>%@</ul>'               => '&lt;ul>%s&lt;/ul>',
+      '<li>%@</li>'               => '&lt;li>%s&lt;/li>',
+
+      '<br>%@'                    => '&lt;br>%s',
+
+      '<div>%@</div>'             => '&lt;div>%s&lt;/div>',
+
+      '<span style="color:#45C1D0">%@</span>' => '&lt;span style="color:#45C1D0">%s&lt;/span>',
+
+      '<p>%@</p>'                 => '&lt;p>%s&lt;/p>',
+      '<p dir="ltr">%@</p>'       => '&lt;p dir="ltr">%s&lt;/p>',
 
       '<a href="target">link</a>'     => '<a href="target">link</a>',
       '<a href="target">"link"</a>'   => '<a href="target">\"link\"</a>',
       '<a href="target"></a>"out"'    => '<a href="target"></a>\"out\"',
       '<a href="http://url.com?param=1&param2=3&param3=%20">link</a>'   =>   '<a href="http://url.com?param=1&param2=3&param3=%20">link</a>',
 
-      '<p>escaped</p><![CDATA[]]>'                => '&lt;p>escaped&lt;/p><![CDATA[]]>',
-      '<![CDATA[]]><p>escaped</p>'                => '<![CDATA[]]>&lt;p>escaped&lt;/p>',
-      '<![CDATA[<p>unescaped</p>]]>'              => '<![CDATA[<p>unescaped</p>]]>',
-      '<![CDATA[<p>unescaped with %@</p>]]>'      => '<![CDATA[<p>unescaped with %s</p>]]>',
-      '<![CDATA[]]><![CDATA[<p>unescaped</p>]]>'  => '<![CDATA[]]><![CDATA[<p>unescaped</p>]]>',
+      '<q>escaped</q><![CDATA[]]>'                => '&lt;q>escaped&lt;/q><![CDATA[]]>',
+      '<![CDATA[]]><q>escaped</q>'                => '<![CDATA[]]>&lt;q>escaped&lt;/q>',
+      '<![CDATA[<q>unescaped</q>]]>'              => '<![CDATA[<q>unescaped</q>]]>',
+      '<![CDATA[<q>unescaped with %@</q>]]>'      => '<![CDATA[<q>unescaped with %s</q>]]>',
+      '<![CDATA[]]><![CDATA[<q>unescaped</q>]]>'  => '<![CDATA[]]><![CDATA[<q>unescaped</q>]]>',
```

Comment:
Curious: why did all of the `p`s here change to `q`?

Reply:
Because now `p` gets the special treatment, so `<p>escaped</p><![CDATA[]]>` will not get escaped, but `<q>escaped</q><![CDATA[]]>` will. This doesn't make any difference when `<p>` is inside CDATA, but still, I felt that those few test lines are meant to be symmetrical, so I used `<q>` everywhere.